### PR TITLE
Flatten github/lab/sourcehut into one ref type

### PR DESCRIPTION
**Pull request by Ben-PH**

### CODE REVIEW

File: src/flakeref.rs [hunk 1]
```diff
@@ -408,25 +398,19 @@ impl FlakeRefType {
...
-                        },
-                        "sourcehut" => FlakeRefType::Sourcehut {
-                            owner,
-                            repo,
-                            ref_or_rev,
-                        },
+                    let src = match flake_ref_type_str {
+                        "github" => GitForge::GitHub,
+                        "gitlab" => GitForge::GitLab,
+                        "sourcehut" => GitForge::SourceHut,
```
Comment (Ben-PH) [inline src/flakeref.rs:404]: I feel this should be rolled up into a from_str trait impl. we could do something like:
```rust
if let Ok(some_gitforge) = ... {
    ...
} else {
    // match on the other possible strings
}
```

...in order to get rid of the unreachable!, but that's probably getting a touch fancy.

**Comment by a-kenji** [comment]
Generally in favor, not sure of the naming `src` here yet.

**Comment by Ben-PH** [comment]
How about `platform`?

**Comment by a-kenji** [comment]
`platform` or `forge` both seem good, I think. `forge` would be a bit double. You decide :).

File: src/flakeref.rs [hunk 2]
```diff
@@ -313,7 +313,7 @@ impl Display for FlakeRefType {
                 write!(f, "{uri}")
             }
             FlakeRefType::GitForge {
-                src,
+                platform: src,
```
Comment (a-kenji) [inline src/flakeref.rs:316]:
```suggestion
                platform,
```
And in the macro as well.

**Review by a-kenji — APPROVED** [review]
Looks Good,
Tests are still pending and please squash the commits here.

File: src/flakeref.rs (hunk 2 above)
Comment (Ben-PH) [reply src/flakeref.rs:316]: dammit: never trust an LSP change, I guess.

Comment (a-kenji) [reply src/flakeref.rs:316]: It probably can't change in the format macro yet, so it tries to stay semantically correct.

File: src/flakeref.rs [hunk 3]
```diff
@@ -301,27 +312,17 @@ impl Display for FlakeRefType {
...
                 write!(f, "{uri}")
             }
-            FlakeRefType::GitHub {
+            FlakeRefType::GitForge {
+                platform,
                 owner,
                 repo,
                 ref_or_rev,
             } => {
+                write!(f, "{platform}:{owner}/{repo}");
```
Comment (a-kenji) [inline src/flakeref.rs:321]: This is much nicer now :+1:.

**Pull request merged**